src/widgets/helper/ChannelView.cpp

```diff
@@ -1439,25 +1461,35 @@ void ChannelView::mouseReleaseEvent(QMouseEvent *event)
     }
     else if (event->button() == Qt::MiddleButton)
     {
-        if (event->screenPos() == this->lastMiddlePressPosition_)
-            this->enableScrolling(event->screenPos());
+        if (this->isScrolling_)
+        {
+            if (event->screenPos() == this->lastMiddlePressPosition_)
+                this->enableScrolling(event->screenPos());
+            else
+                this->disableScrolling();
+
+            return;
+        }
         else
-            this->disableScrolling();
+        {
+            const MessageLayoutElement *hoverLayoutElement =
+                layout->getElementAt(relativePos);
+
+            if (hoverLayoutElement == nullptr ||
+                hoverLayoutElement->getLink().isUrl() == false)
+            {
+                return;
+            }
```

Comment:
`layout` is not guaranteed to exist yet, because we haven't checked the `foundMessage` variable yet. I'd probably just check the `foundMessage` value here instead of moving code around